src/w_wad.c

```diff
@@ -91,90 +92,112 @@ static int *handles = NULL;
 
 static void W_AddFile(const char *name) // killough 1/31/98: static, const
 {
-  wadinfo_t   header;
-  lumpinfo_t* lump_p;
-  unsigned    i;
-  int         handle;
-  int         length;
-  int         startlump;
-  filelump_t  *fileinfo, *fileinfo2free=NULL; //killough
-  filelump_t  singleinfo;
-  boolean     is_single = false;
-  char        *filename = strcpy(malloc(strlen(name)+5), name);
-
-  NormalizeSlashes(AddDefaultExtension(filename, ".wad"));  // killough 11/98
-
-  // open the file and add to directory
-
-  if ((handle = M_open(filename,O_RDONLY | O_BINARY)) == -1)
+    wadinfo_t header;
+    filelump_t *fileinfo;
+    int handle;
+    int length;
+    int startlump;
+
+    // open the file and add to directory
+
+    handle = M_open(name, O_RDONLY | O_BINARY);
+
+    if (handle == -1)
     {
-      if (strlen(name) > 4 && !strcasecmp(name+strlen(name)-4 , ".lmp" ))
-	{
-	  free(filename);
-	  return;
-	}
-      // killough 11/98: allow .lmp extension if none existed before
-      NormalizeSlashes(AddDefaultExtension(strcpy(filename, name), ".lmp"));
-      if ((handle = M_open(filename,O_RDONLY | O_BINARY)) == -1)
-	I_Error("Error: couldn't open %s\n",name);  // killough
+        if (M_StringCaseEndsWith(name, ".lmp"))
+        {
+            return;
+        }
+        I_Error("Error: couldn't open %s", name); // killough
     }
 
-  I_Printf(VB_INFO, " adding %s",filename);   // killough 8/8/98
-  startlump = numlumps;
+    I_Printf(VB_INFO, " adding %s", name); // killough 8/8/98
+
+    startlump = numlumps;
 
-  // killough:
-  if (strlen(filename)<=4 || strcasecmp(filename+strlen(filename)-4, ".wad" ))
+    boolean is_single = false;
+
+    // killough:
+    if (!M_StringCaseEndsWith(name, ".wad"))
     {
-      // single lump file
-      fileinfo = &singleinfo;
-      singleinfo.filepos = 0;
-      singleinfo.size = LONG(W_FileLength(handle));
-      ExtractFileBase(filename, singleinfo.name);
-      numlumps++;
-      is_single = true;
+        // single lump file
+        fileinfo = calloc(1, sizeof(*fileinfo));
+        fileinfo[0].filepos = 0;
```

Comment:
`calloc()` already initializes all fields.

Reply:
Done in https://github.com/fabiangreffrath/woof/pull/1578/commits/c857cb9ddca321a2f90486d9ce484359af58428e